spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+describe "Profile", type: :request do
+  describe "#index" do
+    context "when user is not logged in" do
+      before { get profile_index_path, params: {uid: "user-id"} }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when user is logged in" do
+      let(:user1) { create(:user1) }
```

Comment:
```suggestion
      let(:user) { create(:user1) }
```

Nitpick: I prefer just `user` as a variable if there's one user to start off with when there's no need to distinguish the differences.